bc_obps/registration/tests/endpoints/test_operations_endpoints.py

```diff
@@ -657,3 +661,213 @@ def test_put_operation_with_no_point_of_contact(self):
+        put_response = TestUtils.mock_put_with_auth_role(
+            self,
+            'industry_user',
+            content_type_json,
+            update.json(),
+            self.endpoint + '/' + str(operation.id) + "?submit=true&form_section=1",
+        )
+
+        assert put_response.status_code == 200
+        assert Operation.objects.count() == 1
+        retrieved_operation = Operation.objects.first()
+        assert retrieved_operation.name == 'Declined Operation Name'
+        assert retrieved_operation.submission_date > datetime.strptime(
+            fake_timestamp_from_past, fake_timestamp_from_past_str_format
+        )
+        # assert that operation's submission_date has been updated to the current time (approximately)
+        assert retrieved_operation.submission_date - datetime.now(timezone.utc) < timedelta(seconds=10)
+        assert retrieved_operation.status == Operation.Statuses.PENDING
+
+    def test_put_operation_one_form_section_at_a_time(self):
+        # test setup
+        operator = operator_baker()
+        operation = operation_baker()
+        operation.operator_id = operator.id
+        operation.save(update_fields=['operator_id'])
+        TestUtils.authorize_current_user_as_operator_user(self, operator)
+
+        # update fields visible in section 1 of form (on frontend)
+        update_from_form_section_1 = OperationUpdateIn(
+            name='New and Improved Legal Name',
+            type='Type A',
+            operator_id=operator.id,
+            documents=[],
+            regulated_products=[],
+            naics_code_id=operation.naics_code_id
+            # reporting_activities=[1],
+        )
+        put_response_1 = TestUtils.mock_put_with_auth_role(
+            self,
+            'industry_user',
+            content_type_json,
+            update_from_form_section_1.json(),
+            self.endpoint + '/' + str(operation.id) + "?submit=false&form_section=1",
+        )
+        assert put_response_1.status_code == 200
+        assert Operation.objects.count() == 1
+        retrieved_op = Operation.objects.first()
+        assert retrieved_op.name == 'New and Improved Legal Name'
+        assert retrieved_op.type == 'Type A'
+
+        # update fields visible in section 2 of form
+        update_from_form_section_2 = OperationUpdateIn(
+            name='This name should not be changed',
+            type='bad type',
+            operator_id=operator.id,
+            naics_code_id=operation.naics_code_id,
+            regulated_products=[],
+            # reporting_activities=[],
+            is_external_point_of_contact=True,
+            external_point_of_contact_first_name='Bart',
+            external_point_of_contact_last_name='Simpson',
+            external_point_of_contact_position_title='Scoundrel',
+            external_point_of_contact_email='bart@email.com',
+            external_point_of_contact_phone_number='+17787777777',
+        )
```

Comment:
I believe in section 2, we only get fields related to the user and contact and not operation-related fields.

Reply:
Skip this!Sorry.